client/core/trade.go

```diff
@@ -746,6 +759,12 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 		}
 	}
 
+	// Check if we need to return the change coin because the order has been
+	// canceled.
+	if t.metaData.Status == order.OrderStatusCanceled && !t.hasUnswappedMatches() {
+		t.wallets.fromWallet.ReturnCoins(asset.Coins{t.change})
```

Comment:
Change coins for executed orders stay locked in my testing of this PR.  So, I believe this needs to unlock coins for executed and revoked orders too as long as they are no longer booked (`t.metaData.Status != order.OrderStatusBooked && !t.hasUnswappedMatches())`)?

Note that unmatched market and immediate limits need their funding coins unlocked in `negotiate` rather than `swapMatches`.